windows/devices/uwb/UwbCxAdapter.cxx

```diff
@@ -0,0 +1,239 @@
+                               data = 12;
+                               break;
+                           }
+                           case (::uwb::protocol::fira::Channel::C13): {
+                               data = 13;
+                               break;
+                           }
+                           case (::uwb::protocol::fira::Channel::C14): {
+                               data = 14;
+                               break;
+                           }
+                           }
+                           builder.Push(std::move(UwbAppConfigurationParameter{ data, service_ddi_map.at(ptag) }));
+                       },
+                       [&builder, ptag](::uwb::protocol::fira::ConvolutionalCodeConstraintLength arg) {
+                           // TODO
+                       },
+                       [&builder, ptag](::uwb::protocol::fira::DeviceRole arg) {
+                           builder.Push(std::move(UwbAppConfigurationParameter{ uint8_t(arg), service_ddi_map.at(ptag) }));
+                       },
+                       [&builder, ptag](::uwb::protocol::fira::MultiNodeMode arg) {
+                           builder.Push(std::move(UwbAppConfigurationParameter{ uint8_t(arg), service_ddi_map.at(ptag) }));
+                       },
+                       [&builder, ptag](::uwb::protocol::fira::PrfMode arg) {
+                           builder.Push(std::move(UwbAppConfigurationParameter{ uint8_t(arg), service_ddi_map.at(ptag) }));
+                       },
+                       [&builder, ptag](::uwb::protocol::fira::RangingMethod arg) {
+                           builder.Push(std::move(UwbAppConfigurationParameter{ uint8_t(arg.ToByte()), service_ddi_map.at(ptag) }));
+                       },
+                       [&builder, ptag](::uwb::protocol::fira::RangingMode arg) {
+                           builder.Push(std::move(UwbAppConfigurationParameter{ uint8_t(arg), service_ddi_map.at(ptag) }));
+                       },
+                       [&builder, ptag](::uwb::protocol::fira::SchedulingMode arg) {
+                           builder.Push(std::move(UwbAppConfigurationParameter{ uint8_t(arg), service_ddi_map.at(ptag) }));
+                       },
+                       [&builder, ptag](::uwb::protocol::fira::StsConfiguration arg) {
+                           builder.Push(std::move(UwbAppConfigurationParameter{ uint8_t(arg), service_ddi_map.at(ptag) }));
+                       },
+                       [&builder, ptag](::uwb::protocol::fira::StsPacketConfiguration arg) {
+                           builder.Push(std::move(UwbAppConfigurationParameter{ uint8_t(arg), service_ddi_map.at(ptag) }));
+                       },
+                       [&builder, ptag](::uwb::UwbMacAddress arg) {
+                           builder.Push(std::move(UwbAppConfigurationParameter{ arg.GetValue(), service_ddi_map.at(ptag) }));
```

Comment:
This won't work here since `UwbAppConfigurationParameter::PropertyT` will be `std::span<const uint8_t>` and the internal representation of the span will be copied to the value instead of the actual data the span refers to (which is what we want).

We'll need to limit the types `PropertyT` to probably satisfy `std::is_standard_layout`, but that can be done later. For now, you need to pass something that directly represents the data in memory sequentially, like `std::array<uint8_t, N>`, then pass its buffer pointer as the value, and specify the size directly.